Add first pattern on base documentation.

MaineC opened the pull request:
This relates to #114

It takes a first stab at writing up the basics of InnerSource project setup starting with adding base documentation.

Essentially this is based on experience shared on our company blog at http://tech.europace.de 

While the components of basic project documentation are mentioned in the InnerSource Checklist book I think it makes sense to also include them in our pattern language so people have one place to look for best practices.

MaineC committed:
Add first pattern on base documentation.

This relates to #114

It takes a first stab at writing up the basics of InnerSource project setup
starting with adding base documentation.

rrrutledge reviewed (approved):
Looks great to me.  Thanks for documenting, Isabel.

MaineC committed:
Add templates for initial project documentation.

These templates are meant to be used as a starting point for creating project
documentation. Likely for smaller projects following standard coding practices
their content can be reduced. The more experience the project has with working
with contributors the more helpful those documents will get.

rrrutledge commented on project-setup/templates/CONTRIBUTING-template.md:
```diff
@@ -0,0 +1,40 @@
+# Contributing to ____
```
This is great, but surely there must be an open-source template that we can link to?  Nothing here looks unique to InnerSource.

MaineC replied:
I haven't looked for contributing templates. For READMEs I usually forward people to 

https://m.dotdev.co/how-to-write-a-readme-that-rocks-bc29f279611a

I wonder if any of the bigger foundations have templates for these things that we can link to.

A ton of wisdom on how to run an open source project is also in https://producingoss.com/en/producingoss.html 

There's also the following from GitHub, which if you click through leads to a more detailed explanation of what belongs into a contributing file.

I'm sure there are others that I'm not aware of. A lot of the things I've written down come from practical experience in open source projects.

MaineC replied:
I have added links to these sites to the pattern itself - also added that often it's a good idea to check what open source projects include in their docs.

lenucksi replied:
I had a bunch of links researched when I wrote mine, maybe I'll find some notes or bookmarks of that.
And yes, there were quite a few very nice things.

rrrutledge commented on project-setup/templates/README-template.md:
```diff
@@ -0,0 +1,73 @@
+contributing to InnerSource projects to have a direct link back to company wide
+information from the technological projects they need for their daily work.
+
+### How to become a Trusted Committer
```
These sections on trusted committer management seem less important than contributing guides - suggest they go under the **Contributing** section.

rrrutledge commented on project-setup/templates/README-template.md:
```diff
@@ -0,0 +1,73 @@
+information is identical for all projects in the organisation so central
+information can be linked to from here. 
+
+## Contributing
```
Could use something about how to get help (chat/mailing-list/etc)?

MaineC replied:
Makes sense. I've added some information on that.

rrrutledge reviewed (commented):
Very nice and useful!

ElisaBaum commented on project-setup/base-documentation.md:
```diff
@@ -0,0 +1,105 @@
+  get setup and get started.
+- Project members are spending a lot of time answering getting started
+  questions.
+- Different teams withing the organization have diverging standards for how to
```
```suggestion
- Different teams within the organization have diverging standards for how to
```

ElisaBaum reviewed (approved):
LGTM

martinklewitz commented on project-setup/templates/README-template.md:
```diff
@@ -0,0 +1,73 @@
+See also [mission statement
+chapter](https://producingoss.com/en/producingoss.html#mission-statement) in
+Producing Open Source Software.
+
```
This could include properties and qualities that want to be achieved in this project. Some of those qualities can be very specific and may be changed over the lifetime of the project. They help identify what the project wants to achieve and how what tradeoffs are chosen. For example "in order to strive for simplicity and performance we may not implement feature request that improve convenience".

MaineC replied:
I believe this goes a bit too far for a mission statement. However I believe this is valuable information to include. I have moved it to the contributing section as it is information that contributors will need to make in informed decision about whether their change will be accepted or not.

Writing this text down, I've come to think that this may well be interesting to users who never thought about contributing as well - it could well help decide whether or not to adopt the software.

Likely the template needs some more user focused information.

martinklewitz reviewed (approved):
Looks good.

MaineC committed:
Update project-setup/base-documentation.md

Co-Authored-By: Elisa Baum <elisabaum@hotmail.com>

MaineC committed:
Added links to existing documentation templates

This adds links to external existing documentation on what information to include in a README.md and a contributing guide. While those links point to help for open source projects, most of the information they provide are relevant to InnerSource projects as well.

MaineC committed:
Refine according to review.

This moves the "becoming a committer" section to the contributing doc. It also
adds a section on getting help. In addition it adds a requirement to make
project design values explicit to the contributing section.

MaineC committed:
Merge branch 'basic-setup' of github.com:MaineC/InnerSourcePatterns into basic-setup

rrrutledge commented on project-setup/base-documentation.md:
```diff
@@ -0,0 +1,113 @@
+of information to include in a CONTRIBUTING.md file in various open source projects. 
+Pages like [how to write a readme that rocks](https://m.dotdev.co/how-to-write-a-readme-that-rocks-bc29f279611a), 
+[Open Source Guide from GitHub](https://opensource.guide/) as well as
+the book [Producing Open Source](https://producingoss.com/en/producingoss.html) all
```
I don't actually see good information on a README at the Producing OSS link. Am I just missing it?

MaineC replied:
There is no chapter on writing a good README. The book doesn't say which form to choose to provide the documentation that is often (but not always) found in or linked to in a README.

However in the Getting Started Chapter (https://producingoss.com/en/producingoss.html#starting-from-what-you-have) it does provide a fairly extensive list of things that fellow host team members, users and contributors will need. I don't expect an InnerSource project to cover all of those aspects in their documentation - I find the list helpful as inspiration for what one could cover.

NewMexicoKid replied:
I would suggest you add this clarification to the producingoss link so people know what to expect and what to look for. E.g., ... as well as the book Producing Open Source (although the book doesn't say which form to choose to provide the documentation often found in a README, the [Getting Started Chapter](https://producingoss.com/en/producingoss.html#starting-from-what-you-have) does provide a fairly extensive list of things that fellow host team members, users and contributors will need).

MaineC committed:
Add section for user level documentation.

NewMexicoKid commented on project-setup/base-documentation.md:
```diff
@@ -0,0 +1,113 @@
+* Some information on which turnaround time to expect for modifications made.
+
+
+There are literally tons of good examples for how to write a README.md and what kind
```
I think there are not _literally_ tons; suggest replacing 'literally tons of' with 'many'

NewMexicoKid commented on project-setup/base-documentation.md:
```diff
@@ -0,0 +1,113 @@
+
+# Forces
+
+- A lack of documentation leads to potential contributors taking a long time to
```
All of these forces are driving one side of the problem (towards writing the documentation). Forces are constraints with trade-offs. The trade-offs here are only implied. E.g., A lack of documentation leads to potential contributors taking a long time to get setup and get started. Producing the documentation needed to help out potential contributors will require an investment of time and effort from the project owner.

I think it would be helpful to examine the list of forces (constraints on the problem and solution) to see what the trade-offs/costs are in each force dimension (which isn't to say that *every* force requires listing the trade-off explicitly, but the main ones should say something about this, IMHO).

NewMexicoKid commented on project-setup/base-documentation.md:
```diff
@@ -0,0 +1,113 @@
+If the explanation of the steps to make a contribution are too involved, create
+a separate **CONTRIBUTING.md** document. This document should answer frequently
+asked questions that contributors have asked in the past. There is no need to
+provide a comprehensive book up front. Rather share information that has proven
```
Suggest adding a comma after Rather.

NewMexicoKid commented on project-setup/base-documentation.md:
```diff
@@ -0,0 +1,113 @@
+Pages like [how to write a readme that rocks](https://m.dotdev.co/how-to-write-a-readme-that-rocks-bc29f279611a), 
+[Open Source Guide from GitHub](https://opensource.guide/) as well as
+the book [Producing Open Source](https://producingoss.com/en/producingoss.html) all
+have valuable information on what kind of information to provide. In addition to that this
```
Suggest adding a comma after "that"
In addition to that, this ...

NewMexicoKid reviewed (changes requested):
This is a nice pattern. You might consider naming it: Provide standard base documentation through a README
I would suggest a little more analysis on the forces as the ones presented make it a no-brainer to create a README; there must be some forces that result in the initial context where there is no standard base documentation for the IS project.

NewMexicoKid commented on project-setup/base-documentation.md:
```diff
@@ -0,0 +1,113 @@
+
+A project is to be shared with others as an InnerSource project. In order for
+others to be able to understand what the project is about as well as how to
+contribute, the project needs to provide some base level documentation.
```
The context reflects the initial state, the one where we are before the solution (the README) is applied. I think there should be mention that not only is there a need for others to be able to understand what the project is about, but that there is no base level documentation (why that situation exists should be addressed in the Forces section).

MaineC committed:
Address comments by @NewMexicoKid

NewMexicoKid reviewed (approved):
Thanks for all the hard work and changes. Looks good!
